tests/Unit/HtmlProcessor/CssVariableEvaluatorTest.php

```diff
@@ -0,0 +1,394 @@
+                'css' => '
+                    body {
+                        width: var(--page-width, calc(100vw - 2 * calc(1rem + 10px)));
+                    }
+                ',
+                'expect' => '<body style="width: calc(100vw - 2 * calc(1rem + 10px));">',
+            ],
+            // Processing through `CssInliner` may result in the quotes being changed.
+            // A direct test with HTML is also performed for the following four scenarios.
+            'fallback value with single-quoted string containing opening parenthesis' => [
+                'css' => '
+                    h1 {
+                        content: var(--main-heading, \'Missing heading :(\');
+                    }
+                ',
+                'expect' => '<h1 style=\'content: "Missing heading :(";\'>',
+            ],
+            'fallback value with single-quoted string containing closing parenthesis' => [
+                'css' => '
+                    h1 {
+                        content: var(--main-heading, \'Missing heading ):\');
+                    }
+                ',
+                'expect' => '<h1 style=\'content: "Missing heading ):";\'>',
+            ],
+            'fallback value with double-quoted string containing opening parenthesis' => [
+                'css' => '
+                    h1 {
+                        content: var(--main-heading, "Missing heading :(");
+                    }
+                ',
+                'expect' => '<h1 style=\'content: "Missing heading :(";\'>',
+            ],
+            'fallback value with double-quoted string containing closing parenthesis' => [
+                'css' => '
+                    h1 {
+                        content: var(--main-heading, "Missing heading ):");
+                    }
+                ',
+                'expect' => '<h1 style=\'content: "Missing heading ):";\'>',
+            ],
+        ];
+    }
+
+    /**
+     * @test
+     *
+     * This test simplifies the provision of test data by using an initial `CssInliner` step on some standard HTML,
+     * so that CSS can be provided as the test data rather than direct HTML.
+     *
+     * @param non-empty-string $css
+     * @param non-empty-string $expectedHtmlFragment
+     *
+     * @dataProvider provideCssUsingVaraiblesAndExpectedHtmlFragmentAfterInliningAndEvaluation
+     */
+    public function replacesReferencedVariableIfDefinedOrNotOtherwise(string $css, string $expectedHtmlFragment): void
+    {
+        $subject = $this->buildSubjectWithCssInlined(self::COMMON_TEST_HTML, $css);
+
+        $htmlResult = $subject->evaluateVariables()->render();
+
+        self::assertStringContainsString($expectedHtmlFragment, $htmlResult);
+    }
+
+    /**
+     * @return array<non-empty-string, array{html: non-empty-string, expect: non-empty-string}>
+     */
+    public function provideHtmlUsingVaraiblesAndExpectedHtmlFragmentAfterInliningAndEvaluation(): array
```

Comment:
Varaibles -> Variables (maybe do a global search'n'replace)

Reply:
Can't believe how many times I made the same typo.  Search and replace found it in three method names (though not any comments or documentation, etc.).